### PR TITLE
some tweaks and fixes for building db

### PR DESCRIPTION
**Pull request by benwr**
Some minor changes I added while building the new version of the database. Only relevant for building a new database (and not for, e.g. using a database you already built).

### CODE REVIEW

File: src/taxonomy_db.rs [hunk 1]
```diff
@@ -154,7 +154,11 @@ fn build_new_db(db: sled::Db, source_path: &Path) -> io::Result<TaxonomyDatabase
     let taxa_map_dir = std::fs::read_dir(taxa_map_dir_path)?;
     let fs_iter = taxa_map_dir.filter_map(|f| {
         let path = f.ok()?.path();
-        if path.to_str().unwrap().ends_with("accession2taxid.gz") {
+        let path_str = path.to_str().unwrap();
+        if path_str.ends_with("accession2taxid.gz")
+            || path_str.ends_with("accession2taxid.FULL.gz")
+            || path_str.ends_with("accession2taxid.EXTRA.gz")
```
Comment (vgel) [inline src/taxonomy_db.rs:160]: Maybe it should just accept all `.gz` or `accession*.gz` files?

**Review by vgel — APPROVED** [review]
LGTM, just a small comment

File: src/taxonomy_db.rs (hunk 1 above)
Comment (benwr) [reply src/taxonomy_db.rs:160]: Unfortunately if you clone the whole FTP site there are some gz files that aren't accession2taxid files :/

**Comment by benwr** [comment]
I'm actually going to push some additional tweaks in a second; there was originally a minor subtle bug in the insertion code.

File: src/taxonomy_db.rs (hunk 1 above)
Comment (vgel) [reply src/taxonomy_db.rs:189]: ah :/

File: src/taxonomy_db.rs [hunk 2]
```diff
@@ -107,22 +108,50 @@ fn read_accessions_to_db(
...
+                    // iteration; if it isn't the last run element we want to skip it.
+                    vec![]
+                } else {
+                    // finally we know that we've achieved a proper transition between two runs.
+                    vec![(last_acc.clone(), *last_taxid), (accession.clone(), taxid)]
+                }
             }
-            _ => {}
+            _ => vec![],
+        };
```
Comment (vgel) [inline src/taxonomy_db.rs:145]: Could this be more clear if it was written in iterator style, maybe filter accessions eq to the previous accession, group by taxid, then write the first/last element of the group?

In general the relation between `last_accession_and_taxid`, `last_insertion`, and the current `pair` / `accession` and `taxid` was a bit confusing at times. The comments were helpful, but the names could be a bit more clear -- `prev_pair` / `prev_accession` / `prev_taxid`, `last_inserted_taxid` (`last_insertion.0` doesn't seem to be used), and `current_pair` / `current_accession` / `current_taxid`, maybe?

**Review by vgel — APPROVED** [review]
Hmm, I'm a little worried about how effectively I can review this since I didn't notice anything before, but it looks OK -- the new loop for inserting accessions into the DB took me some time to wrap my head around, but the comments were helpful.

Maybe it would be worthwhile to write a few unit tests for some of the hairier functions here?

Anyways, I'll approve because I think it look OK, regardless.

File: src/taxonomy_db.rs (hunk 2 above)
Comment (benwr) [reply src/taxonomy_db.rs:145]: Yeah, that's a really good point re iterator style. Embarrassed I didn't think of it!

**Pull request merged**